### PR TITLE
Add pairwise ds

### PR DESCRIPTION
**Pull request by martin-carrasco**
Adds the pairwise comparison construction of the dataset.

MISSING:

Make the transforms / labels work with this!

### CODE REVIEW

File: mantra/datasets/pairwise.py [hunk 1]
```diff
@@ -1,25 +1,16 @@
...
-import typing
-from typing import Literal, Tuple, TypeAlias
+from typing import Literal, Tuple
 from tqdm import tqdm
 
 import torch
 
 from mantra.datasets import ManifoldTriangulations
 
 from torch_geometric.data import InMemoryDataset, Data
```
[the remark this answers is not in the document]
Comment (martin-carrasco) [reply mantra/datasets/pairwise.py]: Defered to #36
